src/main/java/carpet/mixins/Player_scarpetEventsMixin.java

```diff
@@ -59,21 +63,26 @@ private void onEntityCollision(Entity entity, CallbackInfo ci)
         }
     }
 
-    @Inject(method = "interactOn", at = @At("HEAD"))
+    @Inject(method = "interactOn", cancellable = true, at = @At("HEAD"))
     private void doInteract(Entity entity, InteractionHand hand, CallbackInfoReturnable<InteractionResult> cir)
     {
         if (!level.isClientSide && PLAYER_INTERACTS_WITH_ENTITY.isNeeded())
         {
-            PLAYER_INTERACTS_WITH_ENTITY.onEntityHandAction((ServerPlayer) (Object)this, entity, hand);
+            if(PLAYER_INTERACTS_WITH_ENTITY.onEntityHandAction((ServerPlayer) (Object)this, entity, hand)) {
+                cir.setReturnValue(InteractionResult.PASS);
```

Comment:
note for the future - there is a difference between PASS / SUCCESS / CONSUME etc. I think events should allow for more than just 'cancel', but that could be specified later.

Reply:
Yeah, might do those too at some point in a seperate PR